### ISSUE TITLE
A notation starting with "{" breaks when I add two new notation starting with "{"

### ISSUE THREAD

**Issue by CohenCyril** [coq/coq#8149]
#### Version
The Coq Proof Assistant, version 8.8.1 (July 2018)
compiled on Jul 25 2018 17:35:15 with OCaml 4.06.0

#### Operating system
Linux 4.16.0-1-amd64 SMP Debian 4.16.5-1 (2018-04-29) x86_64 GNU/Linux

#### Description of the problem

A notation starting with `{` breaks when I add two new notation starting with `{` (adding only one notation is not enough).

I did not manage to remove the dependency in mathcomp though...
```coq
From mathcomp Require Import ssreflect ssrfun ssrbool eqtype choice ssralg.

Notation "{ 'foo' }" := tt (at level 2, format "{ 'foo' }") : ring_scope.

Variable (R : ringType).
Check {rmorphism R -> R}%R. (* success *)

Notation "{ 'bar' }" := tt (at level 2, format "{ 'bar' }") : ring_scope.

Check {rmorphism R -> R}%R.
(* Error: Syntax error: 'bar' '}' expected after '{' (in [constr:operconstr]). *)
```

**Comment by CohenCyril** [comment]
(lowering the levels to `0` solves the problem)

**Comment by herbelin** [comment]
Hi @cohencyril, the notations with pairs of curly brackets are indeed a bit special. If you do `Print Grammar constr`, you shall see that there is a predefined rule `"{" x "}"` at level 0 with `x` at level 99 (see the reference manual), and this is why putting your notation at level 0 works. The motivation for this particular case was to support the notations for  nestings of `sumbool`, `sumor`, `sig`, `sum`.

So, yes, any notation involving `{` and `}` should take care of the presence of the predefined rule, sorry about that. I don't know well how it could be circumvented without a change of the notations for `sumbool` and cie. See also #306 for a related discussion.

Maybe there should be a check when the user tries to define a `{` `}` notation at a level different than 0 since, being delimited on both sides, there is anyway no point to define it at a higher level. So we can maybe let this open until someone has the time to add such check.

**Comment by CohenCyril** [comment]
> Maybe there should be a check when the user tries to define a { } notation at a level different than 0 since, being delimited on both sides, there is anyway no point to define it at a higher level. So we can maybe let this open until someone has the time to add such check.

Indeed, I did not put the level `2` intentionally, I really meant `0`. And I guess a warning would have prevented me from doing this mistake... But I was mostly surprised by the behavior of the system.
